### PR TITLE
Add pre-commit config and configure autoformating tools and pytest from pyproject.toml

### PR DESCRIPTION
**Pull request by GeorgianaElena**

### CODE REVIEW

**Review by consideRatio — APPROVED** [review]
LGTM but I'd say we should hold off merging this a bit until other open PRs get merged to avoid merge conflicts.

- #154
- #156

I see pre-commit.ci is already configured to run for this repo at https://github.com/jupyterhub/traefik-proxy/pull/156 now btw, so the work for this PR can be as simple as `git checkout main`, `git pull`, `git checkout autoformatting`, git `rebase -i main` and then delete the `pick <commit that ran pre-commit>`, `git push --force`

I mared this as draft to signal its good to go - but pending.

File: jupyterhub_traefik_proxy/_version.py
Comment (minrk) [inline jupyterhub_traefik_proxy/_version.py:1]: As an autogenerated file, this should be excluded from auto formatters (or move away from versioneer)

**Review by minrk — APPROVED** [review]
One little comment about excluding versioneer's file from autoformatters (feel free to switch to tbump instead, if you want), but should be ready now

File: jupyterhub_traefik_proxy/_version.py
Comment (consideRatio) [reply jupyterhub_traefik_proxy/_version.py:1]: I opened https://github.com/jupyterhub/jupyterhub-idle-culler/issues/55

**Pull request merged**
